network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -164,60 +168,107 @@ private boolean isSwitchOperableAndPresent(Edge edge, Predicate<Switch> openOper
         }
     }
 
-    private boolean connectNodeBreaker(Resource<VoltageLevelAttributes> voltageLevelResource, Predicate<Switch> isTypeSwitchToOperate) {
+    /**
+     * Check if a switch from the edge is open but cannot be operated (according to the given predicate)
+     * @param edge the edge to test
+     * @param isSwitchOperable the predicate defining if a switch can be operated
+     * @return <code>true</code> if the switch is open and cannot be operated
+     */
+    private boolean checkNonClosableSwitch(Edge edge, Predicate<? super SwitchImpl> isSwitchOperable) {
+        if (edge.getBiConnectable() instanceof SwitchAttributes switchAttributes) {
+            // Get the switch behind the switchAttributes
+            Optional<SwitchImpl> sw = index.getSwitch(switchAttributes.getResource().getId());
+
+            // Test if the switch can be opened according to the predicate
+            return sw.isPresent() && SwitchPredicates.IS_OPEN.test(sw.get()) && isSwitchOperable.negate().test(sw.get());
+        }
+        return false;
```

Comment:
I tried but it would mean having the Optional<SwitchImpl> as method parameter, which is not recommended.

Reply:
```java
    private boolean testSwitchFromEdge(Edge edge, Predicate<Switch> predicate) {
        if (edge.getBiConnectable() instanceof SwitchAttributes switchAttributes) {
            Optional<SwitchImpl> sw = index.getSwitch(switchAttributes.getResource().getId());
            return sw.isPresent() && predicate.test(sw.get());
        }
        return false;
    }

    private boolean checkNonClosableSwitch(Edge edge, Predicate<Switch> isSwitchOperable) {
        return testSwitchFromEdge(edge,
                sw -> SwitchPredicates.IS_OPEN.test(sw) &&
                        isSwitchOperable.negate().test(sw));
    }

    private boolean isAnOpenSwitch(Edge edge) {
        return testSwitchFromEdge(edge, SwitchPredicates.IS_OPEN);
    }
```
This would not work?

Reply:
Good point, we even already have such method so I reused it